Sources/StreamChat/Controllers/DatabaseObserver/StateLayerDatabaseObserver.swift

```diff
@@ -107,16 +114,23 @@ extension StateLayerDatabaseObserver where ResultType == ListResult {
         return collection
     }
     
+    /// Starts observing the database and dispatches changes on the MainActor.
+    ///
+    /// - Important: Does not send the initial value, use the ``item`` for setting the initial value.
     func startObserving(didChange: @escaping (StreamCollection<Item>) async -> Void) throws {
-        try startObserving(didChange: { items in Task(priority: .high) { await didChange(items) } })
+        try startObserving(onContextDidChange: { items in Task.mainActor { await didChange(items) } })
```

Comment:
can you expand a bit more on this change?

Reply:
95% of cases the observer's callback is forwarded to `XYZState` which runs the mutation on the main actor. With the old code what happened was:

1. We are on the NSManagedObjectContext's thread, we run the closure
2. The closure creates a Task and starts running it on the global executor aka background thread
3. didChange is called on the global executor which ends up calling some function in the `XYZState` which requires main actor and the handling jumps to the main thread

With this change, we skip the global executor context switch. Just an optimization.

In rare cases, I want to run additional code in the didChange before forwarding the data to `XYZState`. For these cases one can use `func startObserving(onContextDidChange` which is synchronous and runs on the NSManagedObjectContext's queue (e.g. I might want to do some additional processing. Example is reading reactions and sorting them on the NSManageObjectContext's queue). The other case is, unit-tests.

Reply:
If it goes through and we introduce `@MainActor` state objects then this actually opens up an easy way to use initial argument in the observer.